mountpoint-s3-client/src/mock_client.rs

```diff
@@ -68,9 +68,7 @@ impl MockClient {
     /// Add an object to this mock client's bucket
     pub fn add_object(&self, key: &str, value: MockObject) -> ETag {
```

Comment:
I dont think I understood it right. In that case, we would need to pass the E-Tag every time we want to construct a Mock Object right? And that E-Tag would also be hard coded?

Reply:
Oh got it. Adding it in the `from_bytes`, `constant`,` ramp` methods. And there getting e-tag from another method which generates an etag and not hardcoded.

Reply:
Or you can hard code it each time you call `from_bytes`/`constant`/`ramp` -- the point is that each test gets to decide what ETag it wants to use.

For example, we might add a default ETag constructor for tests that don't care:
```rust
impl ETag {
    #[cfg(test)]
    fn for_tests() -> Self {
        Self { etag: "etag".to_string() }
    }
}
```
and then those tests that don't care would just do:
```rust
client.add_object("key", MockObject::constant(0x00, 1024, ETag::for_tests());
```

But some tests do care about the ETag. For example, we might want to write a test that checks we track the right ETag for each object. So we'd create two objects, and use a different ETag for each of the two calls to `MockObject::constant` so that we can tell them apart.